compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1014,6 +1085,121 @@ public BLangNode transform(SimpleNameReferenceNode simpleNameRefNode) {
         return createTypeNode(simpleNameRefNode.name());
     }
 
+    @Override
+    public BLangNode transform(TupleTypeDescriptorNode typeDescriptorNode) {
+        BLangTupleTypeNode tupleTypeNode = (BLangTupleTypeNode) TreeBuilder.createTupleTypeNode();
+        if (typeDescriptorNode.restTypeDesc() != null) {
+            tupleTypeNode.restParamType = createTypeNode(typeDescriptorNode.restTypeDesc());
+        }
+        for (int i = 0; i < typeDescriptorNode.memberTypeDesc().size(); i++) {
+            if (typeDescriptorNode.memberTypeDesc().get(i).kind() == SyntaxKind.COMMA_TOKEN) {
+                continue;
+            }
+            tupleTypeNode.memberTypeNodes.add(createTypeNode(typeDescriptorNode.memberTypeDesc().get(i)));
```

Comment:
Good to break this in to two statements, for readability.

There are several other places as well.

Reply:
Changed